x/contracts/wasmlanche/src/context.rs

```diff
@@ -334,6 +338,48 @@ impl Context {
     }
 }
 
+pub struct CallContractBuilder<'a> {
+    ctx: &'a mut Context,
+    address: Address,
+    max_units: Gas,
+    value: Option<u64>,
+}
+
+impl<'a> CallContractBuilder<'a> {
+    fn with_address(ctx: &'a mut Context, address: Address) -> Self {
+        CallContractBuilder {
+            ctx,
+            address,
+            max_units: Gas::PassAll,
+            value: None,
+        }
+    }
+
+    pub fn with_max_units(mut self, units: u64) -> Self {
```

Comment:
It would be great if this function could only accept units that are > 0, there might be some semantic changes to do

Reply:
Otherwise, replicate the `Gas` enum type on the go side too so that when it's the `Unit` variant then it's only interpreted as "pass x units" even when it's 0 to avoid any runtime panic because of the conversion from a `u64` to a `NonZeroU64`.